packages/delegation-process/src/services/delegationConsumerService.ts

```diff
@@ -91,6 +104,73 @@ export function delegationConsumerServiceBuilder(
 
       return delegation;
     },
+    async approveConsumerDelegation(
+      delegationId: DelegationId,
+      { logger, correlationId, authData }: WithLogger<AppContext>
+    ): Promise<void> {
+      const delegateId = unsafeBrandId<TenantId>(authData.organizationId);
+
+      logger.info(
+        `Approving delegation ${delegationId} by delegate ${delegateId}`
+      );
+
+      const { data: delegation, metadata } = await retrieveDelegationById(
+        readModelService,
+        delegationId
+      );
+
+      assertDelegationKindIs(delegationKind.delegatedConsumer, delegation);
```

Comment:
Since `retrieveDelegationById` is used in both delegations, how about we add the kind directly there?

Doing so we could filter at the beginning what kind of delegation we need, simplifying the various checks; if not found the error would still be `delegationNotFound`. WYT?

Reply:
Makes sense! Addressed in https://github.com/pagopa/interop-be-monorepo/pull/1210/commits/0c9624bae801efb51ccae58955b2aa9a97fdded7

Let me know what you think